apps/contracts/contracts/EcoEarn.sol

```diff
@@ -106,21 +121,20 @@ contract EcoEarn is AccessControl {
         // Decrease the rewards left
         rewardsLeft[getCurrentCycle()] -= amount;
 
-        // Transfer the reward to the participant
-        require(token.transfer(participant, amount));
+        // Transfer the reward to the participant, will revert if the transfer fails
+        x2EarnRewardsPoolContract.distributeReward(appId, amount, participant, '');
```

Comment:
Shall we also specify that last param is a string for the proof and optional? Pointing to info on how the proof should be formatted

Reply:
Sure!

Reply:
Done